src/drivers/fluid_winmidi.c

```diff
@@ -145,27 +209,107 @@ fluid_winmidi_callback(HMIDIIN hmi, UINT wMsg, DWORD_PTR dwInstance,
     }
 }
 
+/**
+ * build a device name prefixed by its index. The format of the returned
+ * name is: dev_idx:dev_name
+ * The name returned is convenient for midi.winmidi.device setting.
+ * It allows the user to identify a device index through its name or vise
+ * versa. This allows the user to specify a multi device name using a list of
+ * devices index (see fluid_winmidi_midi_driver_settings()).
+ *
+ * @param dev_idx, device index.
+ * @param dev_name, name of the device.
+ * @return the new device name (that must be freed when finish with it) or
+ *  NULL if memory allocation error.
+ */
+static char *fluid_winmidi_get_device_name(int dev_idx, char *dev_name)
+{
+    char *new_dev_name;
+
+    int i =  dev_idx;
+    size_t size = 0; /* index size */
+
+    do
+    {
+        size++;
+        i = i / 10 ;
+    }
+    while(i);
+
+    /* index size + separator + name length + zero termination */
+    new_dev_name = FLUID_MALLOC(size + 2 + FLUID_STRLEN(dev_name));
+    if(new_dev_name)
+    {
+        /* the name is filled if allocation is successful */
+        FLUID_SPRINTF(new_dev_name, "%d:%s", dev_idx, dev_name);
+    }
+    else
+    {
+        FLUID_LOG(FLUID_ERR, "Out of memory");
+    }
+    return new_dev_name;
+}
+
+/*
+  Internal multi device name template.
+*/
+#define MULTI_DEV_PREFIX_LEN 6  /*  length of prefix 'multi:' (6 caracters)*/
+const static char *multi_dev_name = "multi:x[ y z ..]";
+
+/*
+ Add setting midi.winmidi.device and midi.winmidi.maxdevices in the settings.
+
+ MIDI devices names are enumerated and added to midi.winmidi.device setting
+ options. Example:
+ 0:Port MIDI SB Live! [CE00], 1:SB PCI External MIDI, default, multi:x[ y z ..]
+
+ Devices name prefixed by index (i.e 1:SB PCI External MIDI) are real devices.
+ "default" name is the default device.
+ "multi:x[ y z ..]" is the multi device naming. Its purpose is to indicate to
+ the user how he must specify a multi device name in the setting.
+ A multi devices name must begin with the prefix 'multi:' followed by the list
+ of real devices index separated by space(s). Example: "multi:5 3 0"
+
+ midi.winmidi.maxdevices setting is the maximum number of devices opened
+ by the driver.
+*/
 void fluid_winmidi_midi_driver_settings(fluid_settings_t *settings)
 {
     MMRESULT res;
     MIDIINCAPS in_caps;
     UINT i, num;
+    /* maximum MIDI devices that the driver must handle */
+    fluid_settings_register_int(settings, "midi.winmidi.maxdevices", 1, 1, 16, 0);
```

Comment:
This new setting needs to be documented in /doc/fluidsettings.xml

Reply:
In fact this new setting could be removed and replaced by a function that parses the `device name` and returns the number
of devices. I will do that in next commits.